File: src/lua/Base64/base64.lua
-- @start
-- @namespace   com.github.mounthuaguo.mongkeyking
-- @version     0.1
-- @name        Base64 Encode
-- @description Base64 encode and decode
-- @author      heramerom
-- @type        action
-- @action      Copy With Base64 Encode
-- @action      Copy With Base64 Decode
-- @action      Replace With Base64 Encode
-- @action      Replace With Base64 Decode
-- @action      Output With Base64 Encode
-- @action      Output With Base64 Decode
-- @require     https://raw.githubusercontent.com/toastdriven/lua-base64/master/base64.lua
-- @end

function inputText() 
    local text = event.selectionModel.selectionText
    if text == nil or text == '' then
        toast.warn('Please select text')
        return nil
    end
    return text
end


if menu == 'Copy With Base64 Encode' then
    local text = inputText()
    if not text then
        return
    end
    clipboard.setContents(require.a.to_base64(text))
    toast.info('Copied!')
end

if menu == 'Copy With Base64 Decode' then
    local text = inputText()
    if not text then
        return
    end
    clipboard.setContents(require.a.from_base64(text))
    toast.info('Copied!')
end

if menu == 'Replace With Base64 Encode' then
    local text = inputText()
    if not text then
        return
    end
    event.document.replaceString(
        action.selectionModel.selectionStart, 
        action.selectionModel.selectionEnd, 
        require.a.to_base64(text))
end

if menu == 'Replace With Base64 Decode' then
    local text = inputText()
    if not text then
        return
    end
    event.document.replaceString(
        action.selectionModel.selectionStart, 
        action.selectionModel.selectionEnd, 
        require.a.grom_base64(text))
end

if menu == 'Output With Base64 Encode' then
    local text = inputText()
    if not text then
        return
    end
    print(require.a.to_base64(text))
end

if menu == 'Output With Base64 Decode' then
    local text = inputText()
    if not text then
        return
    end
    print(require.a.from_base64(text))
end

File: src/js/Base64/base64.js
// @start
// @namespace com.github.mounthuaguo.mongkeyking.js
// @version 0.1
// @name Base64 Decode and Encode
// @type action
// @action Base64 Encode
// @action Base64 Decode
// @end 


(function () {
    var Base64 = Java.type('java.util.Base64')
    var String = Java.type('java.lang.String')

    var input = event.selectionModel().selectedText()
    if (!input) {
        var str = event.document().scanString()
        if (str.start != -1) {
            input = str.value
        }
    }
    if (!input) {
        toast.warn('Please select some text!')
        return
    }

    if (menu === 'Base64 Encode') {
        var decode = Base64.getEncoder().encodeToString(input.getBytes())
        print('Base64.encode(' + input + '): ')
        print(decode)
        print('')
        return
    }

    if (menu === 'Base64 Decode') {
        var decode = new String(Base64.getDecoder().decode(input))
        print('Base64.decode(' + input + '): ')
        print(decode)
        print('')
        return
    }
})()